src/Service/MagicLinks/Models/SendMagicLinkRequest.cs

```diff
@@ -0,0 +1,30 @@
+using System.ComponentModel.DataAnnotations;
+using System.Net.Mail;
+using Passwordless.Service.MagicLinks.Validation;
+
+namespace Passwordless.Service.MagicLinks.Models;
+
+public class SendMagicLinkRequest
+{
+    [Required]
+    [EmailAddress]
+    public string EmailAddress { get; init; }
+
+    [Required]
+    [MagicLinkTemplateUrl]
+    public string LinkTemplate { get; init; }
```

Comment:
Rename to `UrlTokenTemplate` to match `MagicLinkDTO`?

Reply:
I'll go the other way in order to keep a cleaner API contract

Reply:
+1 for LinkTemplate

Reply:
OR: Just `Url` and we either replace a <token> value, if not found we'll add querystring value automatically. So we won't need to confuse developers with the word "template".

Reply:
I think it's nice that `Template` is in the name, because it's not an actual link. Also, I'd probably call it `UrlTemplate` because "link" is a more broad concept (i.e. technically, this is a hyperlink reference), but I doubt it will cause any confusion.